Formula/o/octave.rb

```diff
@@ -60,9 +59,7 @@ class Octave < Formula
   depends_on "qhull"
   depends_on "qrupdate"
   depends_on "qscintilla2"
-  # Stuck on qt@5
-  # https://octave.discourse.group/t/transition-octave-to-qt6/3139/15
-  depends_on "qt@5"
+  depends_on "qt"
```

Comment:
Going to have to look into this a bit as our Qt6 is missing pkg-config files.

Reply:
Sadly Qt only provides .pc files on Linux. It's a bit tricky to generate full set of `QT_CPPFLAGS` and `QT_LIBS` due to recursive requirements and nested directory structure. Will switch back to Qt5 until this is figured out on either end (i.e. Qt supports macOS pkg-config files or GNU Octave supports CMake builds, neither seems likely).

Reply:
This was merged with the broken Qt support. So, users currently don't have a GUI:
https://savannah.gnu.org/bugs/?65481

See also this previous attempt which looks like it could work now with Octave 9:
https://github.com/Homebrew/homebrew-core/pull/141675/files

Reply:
I was trying something similar locally (parsing the libs out of `configure` and building up `-I` and `-framework`) but I think the indirect requirements cause issues.

I am trying another workaround but in Qt itself to generate some unofficial pkg-config files (https://github.com/Homebrew/homebrew-core/pull/166463). No idea if it will work.

Otherwise, can migrate only Linux to Qt6 and use Qt5 on macOS with a `qscintilla2-qt5` formula.

Reply:
> I am trying another workaround but in Qt itself to generate some unofficial pkg-config files (https://github.com/Homebrew/homebrew-core/pull/166463).

That would be perfect. 🙏
Currently, we are hardcoding the flags for Qt6 in the CI using Homebrew:
https://github.com/gnu-octave/octave/blob/2e5b29aff6ac54472bcecf4b6245616555aabb81/.github/workflows/make.yaml#L431-L435

Actually having pkg-config file for Qt6 on macOS would make that a lot easier. 👍

Reply:
My version of hardcoded Qt flags: #166551

`pkg-config` files would be cleaner though.